src/python/pants/binaries/binary_tool.py

```diff
@@ -78,13 +77,25 @@ def select(self, context=None):
         version = old_opts.get(self.replaces_name)
     return self._select_for_version(version)
 
+  @memoized_property
+  def _binary_util(self):
+    return BinaryUtil.Factory.create()
+
+  @classmethod
+  def get_support_dir(cls):
+    return 'bin/{}'.format(cls.get_name())
+
   @memoized_method
   def _select_for_version(self, version):
-    return BinaryUtil.Factory.create().select(
-      self.support_dir, version, self._get_name(), self.platform_dependent)
+    return self._binary_util.select(
+      supportdir=self.get_support_dir(),
+      version=version,
+      name=self.get_name(),
+      platform_dependent=self.platform_dependent,
+      archive_type=self.archive_type)
 
   @classmethod
-  def _get_name(cls):
+  def get_name(cls):
```

Comment:
Why?

Reply:
No reason at all -- I intuited from a comment you made earlier that said `get_name()` instead of `_get_name()` that that's what you were thinking, and I had the vague feeling that methods that are allowed to be overridden by subclasses are generally named without a leading underscore, based on ~0 python experience. If it's more idiomatic to have the underscore I'll totally put it back.

Reply:
The underscore is a convention for "this is private". Nothing in the language prevents you from calling it from outside, it's just a convention, but it's a useful one nonetheless. And I don't see any need for this to be overridden.  So I think it should be "private", for now.  It's much easier to make something public later, than vice versa.

Reply:
Ah, in haste I was confusing `_get_name()` with `get_support_dir()`. I've made the change in 0dacd36.